dspace-server-webapp/src/test/java/org/dspace/app/rest/ResearcherProfileRestRepositoryIT.java

```diff
@@ -0,0 +1,1093 @@
+            .andExpect(status().isForbidden());
+
+        getClient(authToken).perform(get("/api/eperson/profiles/{id}/eperson", id))
+            .andExpect(status().isForbidden());
+
+    }
+
+    /**
+     * Verify that the createAndReturn endpoint create a new researcher profile.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testCreateAndReturn() throws Exception {
+
+        String id = user.getID().toString();
+        String name = user.getName();
+
+        String authToken = getAuthToken(user.getEmail(), password);
+
+        getClient(authToken).perform(post("/api/eperson/profiles/")
+            .contentType(MediaType.APPLICATION_JSON_VALUE))
+            .andExpect(status().isCreated())
+            .andExpect(jsonPath("$.id", is(id)))
+            .andExpect(jsonPath("$.visible", is(false)))
+            .andExpect(jsonPath("$.type", is("profile")))
+            .andExpect(jsonPath("$", matchLinks("http://localhost/api/eperson/profiles/" + id, "item", "eperson")));
+
+        getClient(authToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isOk());
+
+        getClient(authToken).perform(get("/api/eperson/profiles/{id}/item", id))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.type", is("item")))
+            .andExpect(jsonPath("$.metadata", matchMetadata("dspace.object.owner", name, id, 0)))
+            .andExpect(jsonPath("$.metadata", matchMetadata("dspace.entity.type", "Person", 0)));
+
+        getClient(authToken).perform(get("/api/eperson/profiles/{id}/eperson", id))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.type", is("eperson")))
+            .andExpect(jsonPath("$.name", is(name)));
+    }
+
+    @Test
+    public void testCreateAndReturnWithPublicProfile() throws Exception {
+
+        configurationService.setProperty("researcher-profile.set-new-profile-private", false);
+        String id = user.getID().toString();
+
+        String authToken = getAuthToken(user.getEmail(), password);
+
+        getClient(authToken).perform(post("/api/eperson/profiles/")
+            .contentType(MediaType.APPLICATION_JSON_VALUE))
+            .andExpect(status().isCreated())
+            .andExpect(jsonPath("$.id", is(id)))
+            .andExpect(jsonPath("$.visible", is(true)))
+            .andExpect(jsonPath("$.type", is("profile")))
+            .andExpect(jsonPath("$", matchLinks("http://localhost/api/eperson/profiles/" + id, "item", "eperson")));
+    }
+
+    /**
+     * Verify that an admin can call the createAndReturn endpoint to store a new
+     * researcher profile related to another user.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testCreateAndReturnWithAdmin() throws Exception {
+
+        String id = user.getID().toString();
+        String name = user.getName();
+
+        configurationService.setProperty("researcher-profile.collection.uuid", null);
```

Comment:
For some reason this line seems odd to me.  Why does creation as an Admin require removing the `collection.uuid`?   

I do think we need to have a test that proves what happens when `collection.uuid` is unset (as these tests all set it by default). But, that doesn't appear to be what this test is doing.